roles/openshift_health_checker/openshift_checks/package_version.py

```diff
@@ -73,54 +78,49 @@ def run(self):
-            return self.openshift_to_ovs_version["3.4"]
+        earliest = min(self.openshift_to_ovs_version)
+        latest = max(self.openshift_to_ovs_version)
+        if openshift_version < earliest:
+            return self.openshift_to_ovs_version[earliest]
+        if openshift_version > latest:
+            return self.openshift_to_ovs_version[latest]
 
-        ovs_version = self.openshift_to_ovs_version.get(str(openshift_version))
-        if ovs_version:
-            return ovs_version
+        ovs_version = self.openshift_to_ovs_version.get(openshift_version)
+        if not ovs_version:
+            msg = "There is no recommended version of Open vSwitch for the current version of OpenShift: {}"
+            raise OpenShiftCheckException(msg.format(".".join(str(comp) for comp in openshift_version)))
 
-        msg = "There is no recommended version of Open vSwitch for the current version of OpenShift: {}"
-        raise OpenShiftCheckException(msg.format(openshift_version))
+        return ovs_version
 
     def get_required_docker_version(self):
-        """Return the correct Docker version for the current OpenShift version.
-        If the OpenShift version is 3.1, ensure Docker version 1.8.
-        If the OpenShift version is 3.2 or 3.3, ensure Docker version 1.10.
-        If the current OpenShift version is >= 3.4, ensure Docker version 1.12."""
-        openshift_version = self.get_openshift_version()
-
-        if float(openshift_version) >= 3.4:
-            return self.openshift_to_docker_version["3.4"]
-
-        docker_version = self.openshift_to_docker_version.get(str(openshift_version))
-        if docker_version:
-            return docker_version
-
-        msg = "There is no recommended version of Docker for the current version of OpenShift: {}"
-        raise OpenShiftCheckException(msg.format(openshift_version))
-
-    def get_openshift_version(self):
-        """Return received image tag as a normalized X.Y minor version string."""
-        openshift_version = self.get_var("openshift_image_tag")
-        if openshift_version and openshift_version[0] == 'v':
-            openshift_version = openshift_version[1:]
-
-        return self.parse_version(openshift_version)
-
-    def parse_version(self, version):
-        """Return a normalized X.Y minor version string."""
-        components = version.split(".")
-        if not components or len(components) < 2:
+        """Return the correct Docker version(s) for the current OpenShift version."""
+        openshift_version = self.get_openshift_version_tuple()
+
+        earliest = min(self.openshift_to_docker_version)
+        latest = max(self.openshift_to_docker_version)
+        if openshift_version < earliest:
+            return self.openshift_to_docker_version[earliest]
+        if openshift_version > latest:
+            return self.openshift_to_docker_version[latest]
+
+        docker_version = self.openshift_to_docker_version.get(openshift_version)
```

Comment:
[get_openshift_version_tuple returns exactly two ints in the tuple.](https://github.com/sosiouxme/openshift-ansible/blob/65583d894be627c020732cb66e8c126db334560c/roles/openshift_health_checker/openshift_checks/package_version.py#L121-L126)

BTW in testing the failure case I needed to use a bogus tuple like that since everything else would be found.

There's now a very similar method in the base class, I have a todo to combine these. Maybe should be called `openshift_minor_version_tuple` or something to be clearest.